go/enclave/txpool/txpool.go

```diff
@@ -33,30 +34,69 @@ type TxPool struct {
 	running      bool
 	stateMutex   sync.Mutex
 	logger       gethlog.Logger
+	validateOnly atomic.Bool
 }
 
 // NewTxPool returns a new instance of the tx pool
-func NewTxPool(blockchain *ethchainadapter.EthChainAdapter, gasTip *big.Int, logger gethlog.Logger) (*TxPool, error) {
+func NewTxPool(blockchain *ethchainadapter.EthChainAdapter, gasTip *big.Int, validateOnly bool, logger gethlog.Logger) (*TxPool, error) {
 	txPoolConfig := ethchainadapter.NewLegacyPoolConfig()
 	legacyPool := legacypool.New(txPoolConfig, blockchain)
 
-	return &TxPool{
+	txp := &TxPool{
 		Chain:        blockchain,
 		txPoolConfig: txPoolConfig,
 		legacyPool:   legacyPool,
 		gasTip:       gasTip,
 		stateMutex:   sync.Mutex{},
+		validateOnly: atomic.Bool{},
 		logger:       logger,
-	}, nil
+	}
+	txp.validateOnly.Store(validateOnly)
+	go txp.start()
+	return txp, nil
+}
+
+func (t *TxPool) ChangeMode(validateOnly bool) {
```

Comment:
Could we call this method `SetValidateMode` or something? The `pool.ChangeMode(true)` usages are hard to read.